Log is misleading when a post fails

Related to/witnessed in #1322 and #1289.

When a message is not posted successfully, the log plugin still logs a message saying the message was posted. If you are very familiar with sr3, you might notice that it says `... posted to a file ...` instead of the correct `... posted to exchange: ... topic: ... a file ...`.

This is really misleading and confusing, especially if you are not very familiar with sr3. When we are grepping through logs, we always search for the keyword `posted`. 

When there is a failure posting, we should not log a message saying the file was posted. 

```
[ERROR] sarracenia.flow post flowCallback plugin <bound method Message.post of <sarracenia.flowcb.post.message.Message object at 0x7fd4d6d40940>> crashed: 'identity'
...
[INFO] sarracenia.flowcb.log after_post posted to a file with baseUrl:
```

vs a successful post to an AMQP broker:

```
[INFO] sarracenia.flowcb.log after_post posted to exchange: xpublic topic: v02.post.20241205 a file with baseUrl:
```

( Including this in this issue as it has to do with logging being misintepreted. )

When a file fails to be `put` through a ftp transfer with a sender, the logging gives a very confusing response. It states that it `fails` but then goes on to confirm success of the transfer. I confirmed that the transfer did in fact fail and the success log is wrong.

```
2024-12-06 16:04:45,324 [ERROR] sarracenia.transfer.ftp put failed to put cmcrfe.2024120600_036 to cmcrfe.2024120600_036: 550 Permission denied.
2024-12-06 16:04:45,325 [DEBUG] sarracenia.transfer.ftp chmod sr_ftp chmod 420 cmcrfe.2024120600_036
2024-12-06 16:04:45,389 [INFO] sarracenia.flow send Sent: /apps/sarra/public_data//20241206/MSC-SCI-CMC-OPS/noaa/cmcrfe.2024120600_036  into
/in/cmcrfe.2024120600_036 0-41063
2024-12-06 16:04:45,389 [DEBUG] sarracenia.flow.sender do processing 1 messages worked!
2024-12-06 16:04:45,389 [DEBUG] sarracenia.flowcb.retry after_work loading from post_retry_001: qty=24 ... got: 0
2024-12-06 16:04:45,389 [INFO] sarracenia.flowcb.log after_work sent ok: /in/cmcrfe.2024120600_036
```

I think the problem is that the try/except statement in here 

https://github.com/MetPX/sarracenia/blob/da9dd3754e983fc52db2d9ed15ea53d29c2410e9/sarracenia/flow/__init__.py#L2656-L2672

we should be checking if the `len_written` is greater then 0, and indicate a failed transfer if it is.

--- a/sarracenia/flowcb/post/message.py
+++ b/sarracenia/flowcb/post/message.py
@@ -33,16 +33,21 @@ class Message(FlowCB):
             self.poster = sarracenia.moth.Moth.pubFactory(props)
 
     def post(self, worklist):
-
-        still_ok = []
+        old_ok = worklist.ok
+        worklist.ok = []
         all_good=True
-        for m in worklist.ok:
-            if all_good and hasattr(self.poster,'putNewMessage') and self.poster.putNewMessage(m):
-                still_ok.append(m)
-            else:
-                all_good=False
+        for m in old_ok:
+            try:
+                if all_good and hasattr(self.poster,'putNewMessage') and self.poster.putNewMessage(m):
+                    worklist.ok.append(m)
+                else:
+                    all_good=False
+                    worklist.failed.append(m)
+            except Exception as e:
+                all_good = False
                 worklist.failed.append(m)
-        worklist.ok = still_ok
+                logger.error(f"failed: {e}")
+                logger.debug("Exception details:", exc_info=True)
 
     def metricsReport(self) -> dict:
         if hasattr(self,'poster') and self.poster:

--- a/sarracenia/flowcb/post/message.py
+++ b/sarracenia/flowcb/post/message.py
@@ -46,7 +46,7 @@ class Message(FlowCB):
             except Exception as e:
                 all_good = False
                 worklist.failed.append(m)
-                logger.error(f"failed: {e}")
+                logger.error(f"crashed: {e}")
                 logger.debug("Exception details:", exc_info=True)
 
     def metricsReport(self) -> dict:

